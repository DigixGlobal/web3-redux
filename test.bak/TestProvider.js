import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import { combineReducers } from 'redux-immutable';
import thunk from 'redux-thunk';

import middleware from '../src/middleware';
import reducer from '../src/reducer';

import Web3ConnectTest from './Web3ConnectTest';

const createStoreWithMiddleware = applyMiddleware(thunk, middleware)(createStore);
const reducers = combineReducers({ web3Redux: reducer });
const store = createStoreWithMiddleware(reducers);

export default class TestProvider extends Component {
  render() {
    return (
      <Provider store={store}>
        <Web3ConnectTest />
      </Provider>
    );
  }
}
