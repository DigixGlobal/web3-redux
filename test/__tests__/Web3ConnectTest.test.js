import React from 'react';
import TestProvider from '../TestProvider';
import renderer from 'react-test-renderer';

const instance = renderer.create(<TestProvider />);
const { triggerMethod } = instance.toJSON().props;

function snapshotTest(getter) {
  if (getter) { triggerMethod((context) => context.setData(getter(context))); }
  expect(instance.toJSON()).toMatchSnapshot();
}

it('initializes correctly', () => {
  snapshotTest(({ props }) => [
    props,
    props.web3s.default,
    props.web3s.default.eth,
    props.web3s.default.version,
    props.web3s.default.net,
  ].map(o => Object.keys(o)));
});

it('gets the correct values', () => {
  return new Promise((resolve) => {
    triggerMethod(({ props }) => {
      // do a series of these...
      resolve(Promise.all([
        props.web3s.default.version.getNode(),
        props.web3s.default.version.getNetwork(),
        // props.web3s.default.eth.getCoinbase(),
        // props.web3s.default.version.getEthereum(),
        // props.web3s.default.version.getWhisper(),
      ]));
    });
  })
  .then(() => snapshotTest(({ props }) => [
    props.web3s.default.version.node(),
    props.web3s.default.version.network(),
    // props.web3s.default.eth.coinbase(),
  ]));
});

// it('deploys the contract', () => {
//   const tree = renderer.create(
//     <TestProvider
//       onMount={() => {
//         return new Promise((resolve) => {
//           return setTimeout(resolve, 1000);
//         });
//       }}
//     />
//   ).toJSON();
//   return tree.props.onMount().then(() => {
//     console.log('deployed!');
//     expect(tree).toMatchSnapshot();
//   });
// });
