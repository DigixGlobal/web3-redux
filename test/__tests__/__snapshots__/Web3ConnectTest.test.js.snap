exports[`test contract methods work 1`] = `
<div
  triggerMethod={[Function]}>
  \"{\\\"test\\\":[\\\"0xec900e654a10aacc2203f8be5b1920dda6d9c2a8\\\",\\\"0x1233000000000000000000000000000000000001\\\"]}\"
</div>
`;

exports[`test deploys contracts 1`] = `
<div
  triggerMethod={[Function]}>
  \"{\\\"test\\\":[\\\"transactionHash\\\",\\\"transactionIndex\\\",\\\"blockHash\\\",\\\"blockNumber\\\",\\\"gasUsed\\\",\\\"cumulativeGasUsed\\\",\\\"contractAddress\\\",\\\"logs\\\"]}\"
</div>
`;

exports[`test deploys contracts 2`] = `
<div
  triggerMethod={[Function]}>
  \"{\\\"test\\\":[\\\"names\\\",\\\"register\\\",\\\"__web3\\\"]}\"
</div>
`;

exports[`test gets the correct web3 method values 1`] = `
<div
  triggerMethod={[Function]}>
  \"{\\\"test\\\":[\\\"EthereumJS TestRPC/v3.0.3/ethereum-js\\\",\\\"0\\\"]}\"
</div>
`;

exports[`test initializes correctly 1`] = `
<div
  triggerMethod={[Function]}>
  \"{\\\"test\\\":[[\\\"web3ReduxStore\\\",\\\"web3s\\\",\\\"status\\\"],[\\\"net\\\",\\\"version\\\",\\\"eth\\\",\\\"__web3\\\"],[\\\"balance\\\",\\\"code\\\",\\\"transactionCount\\\",\\\"storageAt\\\",\\\"syncing\\\",\\\"coinbase\\\",\\\"mining\\\",\\\"hashrate\\\",\\\"gasPrice\\\",\\\"accounts\\\",\\\"blockNumber\\\",\\\"block\\\",\\\"blockTransactionCount\\\",\\\"uncle\\\",\\\"transactionFromBlock\\\",\\\"transaction\\\",\\\"transactionReceipt\\\",\\\"getBalance\\\",\\\"getCode\\\",\\\"getTransactionCount\\\",\\\"getStorageAt\\\",\\\"getSyncing\\\",\\\"getCoinbase\\\",\\\"getMining\\\",\\\"getHashrate\\\",\\\"getGasPrice\\\",\\\"getAccounts\\\",\\\"getBlockNumber\\\",\\\"getBlock\\\",\\\"getBlockTransactionCount\\\",\\\"getUncle\\\",\\\"getTransactionFromBlock\\\",\\\"getTransaction\\\",\\\"getTransactionReceipt\\\",\\\"sendTransaction\\\",\\\"sendRawTransaction\\\",\\\"waitForMined\\\",\\\"contract\\\"],[\\\"node\\\",\\\"network\\\",\\\"ethereum\\\",\\\"whisper\\\",\\\"getNode\\\",\\\"getNetwork\\\",\\\"getEthereum\\\",\\\"getWhisper\\\"],[\\\"listening\\\",\\\"peerCount\\\",\\\"getListening\\\",\\\"getPeerCount\\\"]]}\"
</div>
`;
